xrayhttp/handler_test.go

```diff
@@ -124,6 +125,93 @@ func TestHandler(t *testing.T) {
 	}
 }
 
+func TestHandler_context_canceled(t *testing.T) {
+	ctx, td := xray.NewTestDaemon(nil)
+	defer td.Close()
+
+	h := Handler(FixedTracingNamer("test"), http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// check optional interface of http.ResponseWriter
+		if _, ok := w.(http.Hijacker); ok {
+			t.Error("want not implement http.Hijacker, but it does")
+		}
+		if _, ok := w.(http.Flusher); !ok {
+			t.Error("want implement http.Flusher, but it doesn't")
+		}
+		if _, ok := w.(http.Pusher); ok {
+			t.Error("want not implement http.Pusher, but it does")
+		}
+
+		// return InternalServerError if context canceled
+		select {
+		case <-r.Context().Done():
+			w.Header().Set("Content-Type", "text/plain")
+			w.WriteHeader(http.StatusInternalServerError)
+			if _, err := w.Write([]byte("canceled")); err != nil {
+				panic(err)
+			}
+			return
+		default:
+		}
```

Comment:
make the test failure directly.

```suggestion
		select {
		case <-r.Context().Done():
		default:
			t.Error("the context should be canceled, but not")
		}
```

it keeps the test simple.